### DISCUSSION TITLE
使用建议

### DISCUSSION THREAD

**Discussion by lengkristy** [netease-im/NIM_Duilib_Framework#375] [General]
可不可以出一个初级使用教程，比如我想自定义控件，好像和原版的duilib有点不一样，我是界面初学者，只会Win32、MFC，主要是做Linux c后端。我想自己实现一个控件，能够渲染pdf，pdf的库我已经做好了，在win32上测试也没问题，使用hdc绘制。但是用duilib有点懵。

**Comment by nmgwddj** [comment]
example 目录中保存了从最基础（basic）到所有控件（controls）到一个简单的浏览器示例代码。https://github.com/netease-im/NIM_Duilib_Framework/tree/master/examples
更丰富的解决方案示例有 NIM PC Demo，大型客户端项目可以参考它来设计：https://github.com/netease-kit/NIM_PC_Demo

**Comment by lengkristy** [comment]
谢谢拉
